gossipd/routing.c

```diff
@@ -389,15 +389,25 @@ static void bfg_one_edge(struct node *node,
 		/* FIXME: Bias against smaller channels. */
 		u64 fee;
 		u64 risk;
+		u64 requiredcap;
 
 		if (node->bfg[h].total == INFINITE)
 			continue;
 
 		fee = connection_fee(c, node->bfg[h].total) * fee_scale;
-		risk = node->bfg[h].risk + risk_fee(node->bfg[h].total + fee,
-						    c->delay, riskfactor);
+		requiredcap = node->bfg[h].total + fee;
+		risk = node->bfg[h].risk +
+		       risk_fee(requiredcap, c->delay, riskfactor);
 
-		if (node->bfg[h].total + fee + risk >= MAX_MSATOSHI) {
+		if (requiredcap > chan->satoshis * 1000) {
+			/* Skip this edge if the channel has insufficient
+			 * capacity to route the required amount */
+			continue;
+		} else if (requiredcap < c->htlc_minimum_msat) {
+			/* Skip a channels if it indicated that it won't route
+			 * the requeuested amount. */
+			continue;
+		} else if (requiredcap + risk >= MAX_MSATOSHI) {
```

Comment:
can you help me understand why risk is added to requiredcap here and not the previous ones?

Reply:
`risk` is our internal "added cost to take into account riskiness of using this channel".  It's not seen externally, so it doesn't effect whether they'll reject the payment as too small/big.

Reply:
it's all good rusty explained this to me

Reply:
`man lightning-getroute` got me sorted. thanks !